zipkin-server/src/main/java/zipkin/server/ZipkinServer.java

```diff
@@ -13,17 +13,70 @@
  */
 package zipkin.server;
 
+import com.nike.riposte.server.Server;
+import com.nike.riposte.server.config.ServerConfig;
+import com.nike.riposte.server.http.Endpoint;
+import com.nike.riposte.server.http.filter.RequestAndResponseFilter;
+import com.nike.riposte.server.http.filter.impl.AllowAllTheThingsCORSFilter;
+
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+
+import java.io.IOException;
+import java.security.cert.CertificateException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 import zipkin.server.brave.BootstrapTrace;
 
+import static java.util.Collections.singletonList;
+
 @SpringBootApplication
 @EnableZipkinServer
 public class ZipkinServer {
 
-  public static void main(String[] args) {
-    new SpringApplicationBuilder(ZipkinServer.class)
+  public static void main(String[] args) throws CertificateException, InterruptedException, IOException {
+    ConfigurableApplicationContext cac = new SpringApplicationBuilder(ZipkinServer.class)
```

Comment:
interesting. yeah right now, you can't easily disable the built-in http collector, plus there's a port dodging concern with the read apis. for example, google's server only does the collector side, so could probably quite simply swap this even using the default port https://github.com/GoogleCloudPlatform/stackdriver-zipkin/blob/master/collector/src/main/java/com/google/cloud/trace/zipkin/EnableStackdriverCollector.java

I'm not sure how much work it would be to make riposte a transport option for spring boot, similar to undertow, which might be a less intrusive way in.